x/profiles/legacy/v6/store.go

```diff
@@ -248,24 +249,48 @@ func convertChainLinkSignatureData(data *codectypes.Any, cdc codec.BinaryCodec)
 	var signatureAny *codectypes.Any
 	switch signature := v5Signature.(type) {
 	case *v5types.SingleSignatureData:
-		v6Signature := &types.SingleSignatureData{Signature: signature.Signature, Mode: signature.Mode}
+		var signingMode types.CosmosSignMode
+		switch signature.Mode {
+		case signing.SignMode_SIGN_MODE_DIRECT:
+			signingMode = types.COSMOS_SIGN_MODE_DIRECT
+		case signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON:
+			signingMode = types.COSMOS_SIGN_MODE_AMINO
+		case signing.SignMode_SIGN_MODE_TEXTUAL:
+			signingMode = types.COSMOS_SIGN_MODE_RAW
+		default:
+			panic(fmt.Sprintf("unsupported signing mode: %s", signature.Mode))
+		}
+
+		v6Signature := types.NewCosmosSingleSignature(signingMode, signature.Signature)
 		signatureAny, err = codectypes.NewAnyWithValue(v6Signature)
 		if err != nil {
 			panic(err)
 		}
 
 	case *v5types.MultiSignatureData:
```

Comment:
Can you add a test for `MultiSignatrueData` migrations?

Reply:
Added it